spec/integration/device_spec.rb

```diff
@@ -0,0 +1,113 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+require 'bolt_spec/conn'
+require 'bolt_spec/files'
+require 'bolt_spec/integration'
+require 'bolt_spec/puppet_agent'
+require 'bolt_spec/run'
+
+describe "devices" do
+  include BoltSpec::Conn
+  include BoltSpec::Files
+  include BoltSpec::Integration
+  include BoltSpec::PuppetAgent
+  include BoltSpec::Run
+
+  let(:modulepath) { File.join(__dir__, '../fixtures/apply') }
+  let(:config_flags) { %W[--format json --nodes #{uri} --password #{password} --modulepath #{modulepath}] + tflags }
+
+  describe 'over ssh', ssh: true do
+    let(:uri) { conn_uri('ssh') }
+    let(:password) { conn_info('ssh')[:password] }
+    let(:tflags) { %W[--no-host-key-check --run-as root --sudo-password #{password}] }
+
+    let(:device_path) { "/tmp/#{SecureRandom.uuid}.json" }
+
+    def agent_version_inventory
+      { 'groups' => [
+        { 'name' => 'agent_targets',
+          'nodes' => [
+            { 'name' => "ssh://#{conn_info('ssh')[:host]}",
+              'alias' => 'puppet_6',
+              'config' => { 'ssh' => { 'port' => '20024' } } }
+          ] }
+      ],
+        'config' => {
+          'ssh' => { 'user' => 'root',
+                     'host-key-check' => false,
+                     'password' => root_password }
+        } }
+    end
+
+    let(:device_inventory) do
+      device_group = { 'name' => 'device_targets',
+                       'nodes' => [
+                         { 'name' => 'fake_device1',
+                           'config' => {
+                             'transport' => 'remote',
+                             'remote' => {
+                               'remote-transport' => 'fake',
+                               'run-on' => 'puppet_6',
+                               'path' => device_path
+                             }
+                           } }
+                       ] }
+      inv = agent_version_inventory
+      inv['groups'] << device_group
+      inv
+    end
+
+    after(:all) do
+      uninstall('puppet_6', inventory: agent_version_inventory)
+    end
+
+    context "when running against puppet 6" do
+      before(:all) do
+        install('puppet_6', inventory: agent_version_inventory)
+        cmd = 'yes | /opt/puppetlabs/puppet/bin/gem uninstall '\
+              '-i /opt/puppetlabs/puppet/lib/ruby/vendor_gems puppet-resource_api'
```

Comment:
It looks like the RSAPI is included as a vendored gem in puppet agent. It might be worth adding a comment or a TODO about removing this setup when the new version is shipped?